### PR TITLE
*: refactor sqlexec to support using current session

### LINKED ISSUE #32233: current interface cannot support exec multiple sqls in one/current sessions with ExecOption
## Enhancement
Now we have `ExecRestrictedStmt` and `ExecRestrictedSQL` interfaces to support executing SQLs with sys-session-pool and specified ExecOption like snapshot. While sometimes we want to execute multiple SQLs in one session or one transaction. `ExecuteInternal` can support but lacks ExecOption. 

So we need `ExecRestrictedStmt` and `ExecRestrictedSQL` to be extended to support using the current session.

### PR DESCRIPTION
**Pull request by chrysan**
### What problem does this PR solve?

Issue Number: close #32233 

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by xhebox — APPROVED** [review]

File: statistics/handle/handle.go [hunk 1]
```diff
@@ -120,7 +120,7 @@ type Handle struct {
 	// idxUsageListHead contains all the index usage collectors required by session.
 	idxUsageListHead *SessionIndexUsageCollector
 
-	// statsLoad is used to load stats concurrently
+	// StatsLoad is used to load stats concurrently
 	StatsLoad StatsLoad
```
Comment (xuyifangreeneyes) [inline statistics/handle/handle.go:124]: Why do we change this?

**Review by xuyifangreeneyes — APPROVED** [review]
LGTM

**Pull request merged**